tsl/src/continuous_aggs/materialize.c

```diff
@@ -793,6 +806,11 @@ materialization_invalidation_log_get_range(int32 materialization_id, Oid type, i
 	return invalidation_range;
 }
 
+/* We already locked the materialization_invalidation_log table in
+ * ShareRowExclusiveLock mode for the Session in the first txn.
```

Comment:
where do we take a session lock for the `materialization_invalidation_log`? The only ones I see we take, and the only ones mentioned in the comment on line 156 are: the materialization table, the partial view and the raw hypertable...

Reply:
fixed the comment. It should be session lock for materialization table.

Reply:
What is a session lock? I guess it is confusing because it sounds like the lock is held for the duration of the session. Is it even possible to hold locks across multiple transactions unless it is an advisory lock? The documentation says: "Once acquired, a lock is normally held until the end of the transaction."

Reply:
the session lock is for the duration of the materialization.